bridge/elrond/client_test.go

```diff
@@ -25,6 +30,70 @@ func (e TransactionError) Error() string {
 	return string(e)
 }
 
+func createMockArguments() ClientArgs {
+	return ClientArgs{
+		Config: bridge.Config{
+			BridgeAddress:        "bridge address",
+			PrivateKey:           "grace.pem",
+			NonceUpdateInSeconds: 1,
+		},
+		Proxy: &mock.ElrondProxyStub{},
+	}
+}
+
+func TestNewClient(t *testing.T) {
+	t.Parallel()
+
+	args := createMockArguments()
+	args.Config.NonceUpdateInSeconds = 0
+	c, err := NewClient(args)
+	require.Nil(t, c)
+	require.Equal(t, ErrInvalidNonceUpdateInterval, err)
+
+	args = createMockArguments()
+	args.Proxy = nil
+	c, err = NewClient(args)
+	require.Nil(t, c)
+	require.Equal(t, ErrNilProxy, err)
+
+	args = createMockArguments()
+	c, err = NewClient(args)
+	require.Nil(t, err)
+	require.NotNil(t, c)
+}
+
+func TestClient_PollsNonce(t *testing.T) {
+	t.Parallel()
+
+	args := createMockArguments()
+	args.Config.NonceUpdateInSeconds = 1
+	numCalled := uint64(0)
+	args.Proxy = &mock.ElrondProxyStub{
+		GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+			atomic.AddUint64(&numCalled, 1)
+
+			return &data.Account{
+				Nonce: atomic.LoadUint64(&numCalled),
+			}, nil
+		},
+	}
+	c, _ := NewClient(args)
+	require.NotNil(t, c)
+
+	// sleep for 3.5 seconds (do not use a "round number" of seconds as it will make the
+	// test flaky)
+	time.Sleep(time.Second*3 + time.Millisecond*500)
```

Comment:
What I did to avoid this in an earlier implementation is create a mock timer, anyway good to know that round numbers will cause flakyness.

Reply:
right, could have used thatmock timer, didn't bother with it as it will just clog the production component. Can be done if it is a must, but I've considered a 7s long-test not to be such an issue.